Cargo.toml

```diff
@@ -37,6 +37,7 @@ members = [
     "utils/fixed_decimal",
     "utils/litemap",
     "utils/pattern",
+    "utils/ucharstrie",
```

Comment:
suggestion: To be consistent with naming, this component should be named just `charstrie` / `CharsTrie`. 

The 'U' prefix convention in C/C++ wasn't necessary and considered less readable, ex: we went with the ICU4J name `CodePointTrie` over the ICU4C `UCPTrie`.

Reply:
I was thinking about it as well. I ended up shyed away from suggesting this and my rationale is that `char` in Rust is `u8`. `UChar` is a unique Unicode type backed by `u16`.
Since this crate operates exclusively on `u16` then `chartrie` may be slightly misleading.

My position here is very weakly held and in particular if we envision switching/extending the crate to support `u8` then I'd be strongly in favor of following your advice.

Reply:
Yes, I was afraid that char would imply `u8` to a lot of people and this data structure is operating on `u16`.

@zbraniecki What do you see as the advantage of using `u8` rather than `u16` when the underlying data structure we're exporting from ICU4C uses `u16`?

Reply:
Just alignment with how Rust thinks about chars. And if we were to explore it I'd expect us to convert the underlying data structure to u8.

Beyond that, maybe there is some performance/memory consideration - @hsivonen @sffc @Manishearth @markusicu ?

Reply:
We have plans to implement `BytesTrie` and that will support `u8`.

Reply:
Perhaps `Char16Trie` ?

@markusicu convinced me last week that it is better to use u16 units rather than u8 units in the trie even if we are dealing with UTF-8 source text.  Basically we can pack a lot more data into the u16 units.